src/components/Dropdown/Dropdown.tsx

```diff
@@ -99,6 +103,25 @@ export const Dropdown: FC<DropdownProps> = ({
         state.setSelectedKey(activeItemId as string);
     }, [activeItemId]);
 
+    const [maxHeight, setMaxHeight] = useState(DEFAULT_DROPDOWN_MAX_HEIGHT);
+
+    useEffect(() => {
+        const updateMaxHeight = () => setMaxHeight(getInnerOverlayHeight(triggerRef));
+        if (isOpen) {
+            updateMaxHeight();
+            if (autoResize) {
+                window.addEventListener("resize", updateMaxHeight);
+            }
+        } else {
+            setMaxHeight(DEFAULT_DROPDOWN_MAX_HEIGHT);
+        }
```

Comment:
Basically we can omit the whole max height logic if  `autoResize === false`

Reply:
Ah I see that's what you meant, I thought you meant just for the window listener